plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Transform/Inline.hs

```diff
@@ -411,6 +432,48 @@ isFirstVarBeforeEffects n t = do
   go (Unknown : _) = False
   go [] = False
 
+{-| Check if all the variable occurrences are "delayed".
+This means that a variable is used in either of the following ways:
+  * inside a 'delay' term
+  * inside a lambda body
+  * inside a case branch
+-}
+isVarDelayed
+  :: forall name uni fun a
+   . (InliningConstraints name uni fun)
+  => name
+  -> Term name uni fun a
+  -> Maybe Bool
```

Comment:
If this function checks `if all the variable occurrences are "delayed"`, then it's better to return `Bool` rather than `Maybe Bool`.

Reply:
I can do it, but this makes the function less general and therefore less potentially applicable outside of the inliner, since the choice of default value is determined by its usage in the inliner.

Reply:
I think that's very premature generalization - there's a good chance it will never be needed outside of the inliner, and even if it is needed by something else, the simplified version may still be sufficient.

Reply:
👌🏼